app.config.js

```diff
@@ -1,13 +1,20 @@
-import { sentryApi } from './src/config/secrets';
+import { secrets } from './src/config/secrets';
 
-export default ({ config }) => ({
-  ...config,
-  hooks: {
-    postPublish: [
-      sentryApi?.config && {
-        file: 'sentry-expo/upload-sourcemaps',
-        config: sentryApi.config
+export default ({ config }) => {
+  const sentryConfig = secrets[config.slug]?.sentryApi?.config;
+  const hooks = sentryConfig
+    ? {
+        postPublish: [
+          {
+            file: 'sentry-expo/upload-sourcemaps',
+            config: sentryConfig
+          }
+        ]
```

Comment:
when reading this and the part in App.js i think the "same" stuff is done way too differently. it could be easier to read and capture if it is more in common, what do you think about:

```
const namespace = config.slug;
const sentryApi = secrets[namespace].sentryApi;
const hooks = sentryApi?.config
  ? {
      postPublish: [
        {
          file: 'sentry-expo/upload-sourcemaps',
          config: sentryApi.config
        }
      ]
```

Reply:
I don't really mind in that case. I updated to your version and pushed it. Still works the same way 🙂